app/views/users/new.html.erb

```diff
@@ -0,0 +1,28 @@
+<h1>Welcome to the Sign Up page!</h1>
+
+<% if @new_user.errors.any? %>
+  <div class="errors">
+    <h2> ERROR!!!! </h2>
+    <ul>
+      <% @new_user.errors.each do |attribute, message| %>
+        <li><strong><%= attribute %></strong>: <%= message %></li>
+      <% end %>
+    </ul>
+  </div>
```

Comment:
It's great to display errors in forms. When the errors are displayed, I noticed that it changes the formatting of the form.

In addition to displaying errors, it can be nice to prevent users from having errors in form by making it clear which fields are required. Rails makes this easy:
`f.text_field :username, required: true`